pkg/apis/config/types.go

```diff
@@ -18,7 +19,7 @@ type Configuration struct {
 	// HealthCheckConfig is the config for the health check controller.
 	HealthCheckConfig *healthcheckconfig.HealthCheckConfig
 	// CosignPublicKeys is the cosign public keys used to verify image signatures.
-	CosignPublicKeys []string
+	CosignPublicKeys *runtime.RawExtension
```

Comment:
Looking at the docs for RawExtension[1], it's advised that the external type is runtime.RawExtension while the internal one is runtime.Object. Would this be applicable in this case?

[1] https://pkg.go.dev/k8s.io/apimachinery/pkg/runtime#RawExtension

Reply:
I am aware of it, and even tried to implement it this way, however I decided not to follow this recommendation because of several reasons

- these types are not real API objects exposed via REST server, they are just used to introduce some well-defined structure of the configuration file
- the external version type is practically not used anywhere in code, it is indirectly used when the config file is read and decoded into the intern version  https://github.com/gardener/gardener-extension-shoot-lakom-service/blob/5de099d748625180681d396531d7f8d3bfb02b7f/pkg/cmd/options.go#L62-L63 
-  if the internal type is using `runtime.Object`, the lakom admission controller config type must follow the same path and be like a k8s API type with conversions, deep copies, defaulting, API scheme, API group, etc. I decided to save this overhead and use a simple go type that can be decoded/encoded from/to yaml file( also json).

Reply:
Makes sense 👍